modules/userdirectory/src/main/java/org/opencastproject/userdirectory/InMemoryUserAndRoleProvider.java

```diff
@@ -124,75 +118,86 @@ protected void activate(ComponentContext cc) {
               + "#                                                    #\n"
               + "######################################################");
     }
-
-    // Create the digest user
-    createSystemUsers();
   }
 
   @Override
   public String getName() {
     return PROVIDER_NAME;
   }
 
+  /**
+   * Get a list of all users for the current organization, loading the users from configuration if the organization had
+   * not been loaded before.
+   *
+   * @return List of users.
+   */
+  private List<User> getOrganizationUsers() {
+    Organization organization = securityService.getOrganization();
+    if (!inMemoryUsers.containsKey(organization)) {
```

Comment:
Hmm... is it possible that this method could be called by multiple threads concurrently even for the same organisation? In that case, this PR would have a thread-safety issue.

@lkiesow?

Reply:
Makes sense. While it's highly unlikely, it is possible. I've fixed that.